File: cli/template/packages/python/tests/test_client.py
"""
Tests for {{Name}}Client.
"""

import pytest
from {{name}}_do import {{Name}}Client


class Test{{Name}}Client:
    """Tests for the {{Name}}Client class."""

    def test_create_client_with_defaults(self) -> None:
        """Test creating a client with default options."""
        client = {{Name}}Client()
        assert client.base_url == "https://{{name}}.do"
        assert client.api_key is None

    def test_create_client_with_api_key(self) -> None:
        """Test creating a client with an API key."""
        client = {{Name}}Client(api_key="test-key")
        assert client.api_key == "test-key"

    def test_create_client_with_custom_url(self) -> None:
        """Test creating a client with a custom base URL."""
        client = {{Name}}Client(base_url="https://custom.{{name}}.do")
        assert client.base_url == "https://custom.{{name}}.do"

File: cli/template/packages/nim/tests/test_client.nim
import unittest
import std/[options]
import {{name}}_do

suite "{{Name}}Client":
  test "creates with default options":
    let client = new{{Name}}Client()
    check client.baseUrl == "https://{{name}}.do"

  test "creates with custom base URL":
    let client = new{{Name}}Client(baseUrl = "https://custom.{{name}}.do")
    check client.baseUrl == "https://custom.{{name}}.do"

  test "accepts API key":
    let client = new{{Name}}Client(apiKey = "test-key")
    check client != nil

  test "creates with options object":
    let opts = new{{Name}}ClientOptions(
      apiKey = some("test-key"),
      baseUrl = "https://test.{{name}}.do"
    )
    let client = new{{Name}}Client(opts)
    check client.baseUrl == "https://test.{{name}}.do"

  test "not connected initially":
    let client = new{{Name}}Client()
    check not client.connected

suite "{{Name}}Error":
  test "creates with message":
    let err = new{{Name}}Error("Something went wrong")
    check err.msg == "Something went wrong"

  test "creates with code":
    let err = new{{Name}}Error("Auth failed", code = some("AUTH_ERROR"))
    check err.code.get == "AUTH_ERROR"
